### PR TITLE
ChibiOS: support STM32H7A3

### PR DESCRIPTION
**Pull request by tridge**
This adds support for a MatekH7A3 board, which uses a H7A3 instead of a H743
The biggest difference with the H7A3 is it has 8k flash sectors. 

This PR builds on the following PRs which should be merged first:
 - https://github.com/ArduPilot/ardupilot/pull/26079
 - https://github.com/ArduPilot/ChibiOS/pull/87

The only thing not working on the MatekH7A3 is microSD support

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/common/stm32h7_A3_mcuconf.h [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+/*
+  this header is modelled on the one for the Nucleo-144 H743 board from ChibiOS
+ */
+#pragma once
+
+// we want to cope with both revision XY chips and newer chips
+#if !defined(HAL_CUSTOM_MCU_CLOCKRATE) || HAL_CUSTOM_MCU_CLOCKRATE <= 400000000
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/common/stm32h7_A3_mcuconf.h:22]: Is this right? I thought it had a lower clock rate. Does it even have X&Y revs?

File: libraries/AP_HAL_ChibiOS/hwdef/common/stm32h7_A3_mcuconf.h [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+ * Reading STM32 Reference Manual is required.
+ */
+#define STM32_LSI_ENABLED                   FALSE
+#define STM32_CSI_ENABLED                   FALSE
+#define STM32_HSI48_ENABLED                 TRUE
+#define STM32_LSE_ENABLED                   FALSE
+#define STM32_HSIDIV                        STM32_HSIDIV_DIV1
+
+/*
+ * Setup clocks for the STM32H70 which vary quite significantly from other H7 variants
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/common/stm32h7_A3_mcuconf.h:103]: comment is wrong

**Review by andyp1per — CHANGES_REQUESTED** [review]
Just a couple of minor changes

File: libraries/AP_HAL_ChibiOS/hwdef/common/stm32h7_A3_mcuconf.h (hunk 1 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/common/stm32h7_A3_mcuconf.h]: cut&paste error, fixed

File: libraries/AP_HAL_ChibiOS/hwdef/common/stm32h7_A3_mcuconf.h (hunk 2 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/common/stm32h7_A3_mcuconf.h]: fixed, thanks

**Review by andyp1per — APPROVED** [review]

**Pull request merged**